pkg/network/http/http_statkeeper.go

```diff
@@ -96,22 +110,38 @@ func (h *httpStatKeeper) handleIncomplete(tx httpTX) {
 	delete(h.incomplete, key)
 }
 
-func (h *httpStatKeeper) newKey(tx httpTX) Key {
-	path := tx.Path(h.buffer)
-	pathString := h.intern(path)
-
+func (h *httpStatKeeper) newKey(tx httpTX, path string) Key {
 	return Key{
 		SrcIPHigh: uint64(tx.tup.saddr_h),
 		SrcIPLow:  uint64(tx.tup.saddr_l),
 		SrcPort:   uint16(tx.tup.sport),
 		DstIPHigh: uint64(tx.tup.daddr_h),
 		DstIPLow:  uint64(tx.tup.daddr_l),
 		DstPort:   uint16(tx.tup.dport),
-		Path:      pathString,
+		Path:      path,
 		Method:    Method(tx.request_method),
 	}
 }
 
+func (h *httpStatKeeper) processHTTPPath(tx httpTX) (pathStr string, rejected bool) {
+	path := tx.Path(h.buffer)
+
+	for _, r := range h.replaceRules {
+		if r.Re.Match(path) {
+			if r.Repl == "" {
+				// this is a "drop" rule
+				return "", true
+			}
+
+			// TODO: consider removing the `break` statement here so rules can be "chained"
+			path = r.Re.ReplaceAll(path, []byte(r.Repl))
+			break
```

Comment:
I think we should remove the `break`. I would expect multiple rules to be applied if they match.

Reply:
ok, I will do that and add a test case covering it